src/net.cpp

```diff
@@ -1871,7 +1902,14 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                 }
             } else {
                 // Not a feeler
-                std::tie(addr, addr_last_try) = addrman.Select();
+                if (preferred_net.has_value()) {
```

Comment:
In `377e81e2e409470d00e69247d0d9e980773ab73e`, I suppose this verification could be avoided. In `Select`, 
`new_only` is false by default and `preferred_net` is optional and it will only have a value when `MaybePickPreferredNetwork`returns true. So, if `preferred_net` doesn't have any value, I suppose that `addrman.Select(false, preferred_net)` and `addrman.Select()` will have the same effect.

```diff
@@ -1901,15 +1915,7 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                     std::tie(addr, addr_last_try) = addrman.Select(true);
                 }
             } else {
-                // Not a feeler
-                if (preferred_net.has_value()) {
-                    // pick an extra outbound peer from a preferred network.
-                    // The eviction logic in net_processing makes sure that
-                    //  a peer from another network will be evicted.
-                    std::tie(addr, addr_last_try) = addrman.Select(false, preferred_net);
-                } else {
-                    std::tie(addr, addr_last_try) = addrman.Select();
-                }
+                std::tie(addr, addr_last_try) = addrman.Select(false, preferred_net);
```